protocol/version.go

```diff
@@ -40,23 +40,21 @@ func IsSupportedVersion(supported []VersionNumber, v VersionNumber) bool {
 }
 
 // HighestSupportedVersion finds the highest version number that is both present in other and in SupportedVersions
-// the versions in other do not need to be ordered
 // it returns true and the version number, if there is one, otherwise false
-func HighestSupportedVersion(other []VersionNumber) (bool, VersionNumber) {
-	var otherSupported []VersionNumber
-	for _, ver := range other {
+func HighestSupportedVersion(ours, theirs []VersionNumber) (bool, VersionNumber) {
+	var theirsSupported []VersionNumber
+	for _, ver := range theirs {
 		if ver != VersionUnsupported {
-			otherSupported = append(otherSupported, ver)
+			theirsSupported = append(theirsSupported, ver)
```

Comment:
This is allocation-heavy without much work from the client, we should benchmark that at some time.